lib/enigma.rb

```diff
@@ -1,8 +1,24 @@
-class Enigma
-  attr_reader :shift_generator, :character_set
+require_relative 'cipher'
+
+class Enigma < Cipher
 
   def initialize
-    @character_set = ("a".."z").to_a << " "
-    @shift_generator = ShiftGenerator.new
+    super
+  end
+
+  def encrypt(message, key = self.default_key, date = self.default_date)
+    shift = input_verification(key, date)
+    { encryption: encrypt_message(message, shift[0]),
+      key: shift[1],
+      date: shift[2]
+    }
```

Comment:
encrypt method calls on methods defined in cipher so that the sole responsibility of encrypt is to take input messages and output messages along with the keys used to encrypt the messages